rfc/2022/2022-11-02_TechTeam4_Preview_Environment_Architecture.md

```diff
@@ -0,0 +1,97 @@
+# RFC: The Preview environment architecture will generate publicly accessible environments hosted in EKS using mocked test data.
+
+* Comment Deadline: `2022-12-15`
+* Team Crew and Name: Platform/Tech Team 4
+* Authors:
+  * [pjhill](https://github.com/pjhill)
+  * [JoeTice](https://github.com/JoeTice)
+* [Original RFC Pull Request](https://github.com/department-of-veterans-affairs/va.gov-platform-arch/pull/-1)
+
+## Background
+Review Instances is the term used for the current approach to the ephemeral environments used on the VA.gov platform. Review instances provide many helpful capabilities to preview products on the platform, but have also been the source of negative feedback from VFS teams over recent quarters. Between platform feedback surveys and recent discovery efforts, a list of pain points was surfaced and is listed below...
+1. VFS team members & VA Stakeholders without SOCKS access or the technical knowledge to use SOCKS can't use review instances
+2. Review Instance creation operates under the deprecated BRD (build, release, deploy) paradigm
+3. Review Instances don't always spin up reliably
+4. Review Instances are only available for a limited time, and therefore aren't available when needed
+5. VFS team members are unable to preview changes from multiple sources in a single preview environment
+6. VFS Teams are unclear on the benefits and intended use of Review Instances
+7. VFS team members are unable to reach review instances with tools hosted in the cloud
+8. Review Instances can't be tested on real or virtual mobile devices
+
+As a result of these issues, review instance usage is low. Review Instances are generated for every PR that is created. As of 11/18/2022, less than 10% of the review instances that are automatically created are ever accessed by a user.
```

Comment:
More commentary than anything... But spin up for every PR feels very wasteful but also seems to have a high probability of really skewing the usage numbers to a point of being meaningless.

Reply:
Yeah, we had a similar thought when we saw the low usage percentage. We attempted to quantify the cost of used vs. unused review instances [here](https://va-gov.domo.com/page/600769572).

Reply:
@little-oddball - Additional info that may be helpful, we discussed the potential for waste internally on the team, which we've begun quantifying in the board Peter shared. When we discussed w/ @raywangoctova, we decided that we would continue to track, and as we understood the cost better, we could evaluate whether the cost was worth the convenience of them being available automatically. Since feature parity is also a preferred goal, this approach aligned with that, while not closing the door for future improvements.